test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -219,6 +219,124 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
 	return nil
 }
 
+// GetOVSPorts returns a list of OVS ports for a given pod.
+func (data *testData) GetOVSPorts(targetPod *testPodInfo) ([]string, error) {
+	cmd := []string{"ip", "addr", "show"}
+	stdout, _, err := data.e2eTestData.RunCommandFromPod(data.e2eTestData.GetTestNamespace(), targetPod.podName, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error listing interfaces for pod %s: %w", targetPod.podName, err)
+	}
+
+	var ovsPorts []string
+	lines := strings.Split(stdout, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) >= 2 && strings.HasSuffix(fields[0], ":") {
+			ifaceName := strings.Split(strings.TrimSuffix(fields[1], ":"), "@")[0]
+			ovsPorts = append(ovsPorts, ifaceName)
+		}
+	}
+	return ovsPorts, nil
+}
+
+// MatchOVSPorts ensures that all ports in afterOVSPorts exist in beforeOVSPorts.
+func (data *testData) CheckOVSPorts(beforeOVSPorts, afterOVSPorts []string) error {
+	portSet := make(map[string]struct{})
+	for _, port := range beforeOVSPorts {
+		portSet[port] = struct{}{}
+	}
+
+	for _, port := range afterOVSPorts {
+		if _, exists := portSet[port]; !exists {
+			return fmt.Errorf("OVS port %s is missing after restart", port)
+		}
+	}
+
+	return nil
+}
+
+// CheckOVSPortsAfterAgentRestart verifies OVS ports before and after restarting Antrea agent pods.
+func (data *testData) CheckOVSPortsAfterAgentRestart(e2eTestData *antreae2e.TestData, t *testing.T) error {
+	var beforeRestartOVSPorts []string
+	for _, pod := range data.pods {
+		ports, err := data.GetOVSPorts(pod)
+		if err != nil {
+			t.Fatalf("Error getting OVS ports before restart: %v", err)
+		}
+		beforeRestartOVSPorts = append(beforeRestartOVSPorts, ports...)
+	}
+
+	if err := e2eTestData.RestartAntreaAgentPods(90 * time.Second); err != nil {
```

Comment:
I meant could we remove some Pods after Agent stops, and then check the secondary OVS ports are removed after Agent restarts.